### PR TITLE
Use warnings module instead of writing directly to stderr.

### LINKED ISSUE #228: Use Python's logging module instead of writing directly to stderr
There are a few occasions where cyvcf2.pyx writes to stderr. It would be nice to be able to capture and/or filter these in some cases. Would it be ok if these were replaced with calls to the [`logging`](https://docs.python.org/3/library/logging.html) module? There seem to be some caveats with using the logger inside cython, but I think it should be ok if logger attributes related to stack frames aren't used.

The specific message I want to silence right now is "no intervals found for ...", because I'm deliberately doing `next(vcf(contig_id))` for each contig id found in the header to see if any variants exist for that contig (I then create a mapping from contig id to vcf file, which is handy for vcfs split by chromosome).
https://github.com/brentp/cyvcf2/blob/2addcaf097bba1b6fa219a260e6ec39a2b2e936c/cyvcf2/cyvcf2.pyx#L403
https://github.com/brentp/cyvcf2/blob/2addcaf097bba1b6fa219a260e6ec39a2b2e936c/cyvcf2/cyvcf2.pyx#L457

### PR DESCRIPTION
**Pull request by grahamgower**
Closes #228.

### CODE REVIEW

File: cyvcf2/cyvcf2.pyx [hunk 1]
```diff
@@ -905,7 +906,7 @@ cdef class VCF(HTSFile):
             nv += 1
             if nv == n_variants:
                 break
-        sys.stderr.write("tested: %d variants out of %d\n" % (nv, nvt))
+        warnings.warn("tested: %d variants out of %d" % (nv, nvt))
```
Comment (grahamgower) [inline cyvcf2/cyvcf2.pyx:909]: Not sure we want to keep this?

File: cyvcf2/cyvcf2.pyx [hunk 2]
```diff
@@ -929,7 +930,7 @@ cdef class VCF(HTSFile):
         for sj in range(ns):
             sample_j = samples[sj]
             if _hets[sj] == 0:
-                print("peddy: no hets found for sample %s\n" % sample_j, file=sys.stderr)
+                warnings.warn("peddy: no hets found for sample %s" % sample_j)
```
Comment (grahamgower) [inline cyvcf2/cyvcf2.pyx:933]: Nor this?

File: cyvcf2/cyvcf2.pyx [hunk 3]
```diff
@@ -410,7 +411,7 @@ cdef class VCF(HTSFile):
                     bcf_destroy(b)
                     break
                 if bcf_subset_format(self.hdr, b) != 0:
-                    sys.stderr.write("could not subset variant")
+                    warnings.warn("could not subset variant")
```
Comment (grahamgower) [inline cyvcf2/cyvcf2.pyx:414]: This needs a test case.

File: cyvcf2/cyvcf2.pyx (hunk 1 above)
Comment (brentp) [reply cyvcf2/cyvcf2.pyx:909]: yes, we can leave this. It's only used by peddy. I wouldn't do it this way again, but let's leave for now.

File: cyvcf2/cyvcf2.pyx (hunk 2 above)
Comment (brentp) [reply cyvcf2/cyvcf2.pyx:933]: same. leave all the peddy stuff in (but changing to warning is great).

File: cyvcf2/cyvcf2.pyx (hunk 3 above)
Comment (grahamgower) [reply cyvcf2/cyvcf2.pyx]: I checked the source of `bcf_subset_format()` and found it always succeeds (returning 0). So I removed the code path and put in an assertion instead.

**Pull request merged**